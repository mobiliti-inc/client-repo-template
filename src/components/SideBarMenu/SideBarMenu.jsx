import { withRouter } from 'react-router';

import styles from './SideBarMenu.scss';
import Link from '../Link/Link';

class SideBarMenu extends React.PureComponent {
	renderMenuContents = (data) => {
		const currentPath = this.props.history.location.pathname;
		return data.map((item, index) => {
			const isActiveClass = currentPath === item.path ? 'side-bar-menu__link-active' : '';
			const paddingBottomClass = data.length !== (index + 1) ? 'side-bar-menu__ladder' : '';
			const isDisabledClass = item.disabled ? 'link-disabled' : '';
			return (
				<div key={item.title} styleName={paddingBottomClass}>
					<div styleName={`side-bar-menu__link-wrapper ${isActiveClass} ${isDisabledClass}`} >
						<Link to={item.path}>{item.title}</Link>
					</div>
				</div>
			);
		});
	}
	render() {
		const { menuContent } = this.props;
		return (
			<div styleName="side-bar-menu">
				{this.renderMenuContents(menuContent)}
			</div>

		);
	}
}

SideBarMenu.propTypes = {
	history: PropTypes.objectOf(PropTypes.any).isRequired,
	menuContent: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default withRouter(CSSModules(SideBarMenu, styles, { allowMultiple: true }));
