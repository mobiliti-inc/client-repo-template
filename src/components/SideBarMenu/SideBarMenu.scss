@import "~variables";

.side-bar-menu {
    width: 100%;
    position: relative;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    @media (max-width: $mobile-width)  {
        box-shadow: none;
	}
    &__link-wrapper {
        width: 100%;
        height: 42px;
        display: flex;
        align-items: center;
        margin: 12px 0;
        border-left: none;
        & > a {
            color: $darker-blue;
            font-size: 20px;
            text-transform: capitalize;
            padding-left: 33px;
            width: 100%;
            font-family: $hanken-sans;
        }
    }
    &__link-active {
        border-left: 4px solid $blue;
    }
    &__ladder {
        border-bottom: 1px solid rgba(0, 0, 0, 0.1);
        width: 100%;
    } 
}

.link-disabled {
    pointer-events: none;
    opacity: 0.6;
}