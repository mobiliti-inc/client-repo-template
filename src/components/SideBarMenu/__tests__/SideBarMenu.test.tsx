import SideBarMenu from '../SideBarMenu';
import sideMenuContentData from '../../../pages/Profile/common/sideMenuContentData';


describe('SideBarMenu', () => {
	const defaultProps = {
		history: jest.fn(),
		menuContent: sideMenuContentData,
	};
	it('renders', () => {
		const tree = shallow(<SideBarMenu {...defaultProps} />);
		matchSnapshot(tree);
	});
});
