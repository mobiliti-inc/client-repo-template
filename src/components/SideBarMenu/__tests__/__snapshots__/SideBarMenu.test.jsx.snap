// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarMenu renders 1`] = `<Route />`;
